pkgs/development/python-modules/tiktoken/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, stdenv
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, rustPlatform
+, setuptools-rust
+, libiconv
+, requests
+, regex
+, blobfile
+}:
+let
+  pname = "tiktoken";
+  version = "0.2.0";
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-30Gj1HhJl1e1sy6uXpdlfPFZ2NnmdkBJ3Xw6u0nhtA8=";
+  };
+  postPatch = ''
+    cp ${./Cargo.lock} Cargo.lock
+  '';
+in
+buildPythonPackage {
+  inherit pname version src postPatch;
+  format = "setuptools";
+
+  disabled = pythonOlder "3.6";
+
+  nativeBuildInput = [
+    setuptools-rust
+  ];
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src postPatch;
+    name = "${pname}-${version}";
+    hash = "sha256-jWgFSxJxdyO/U7kfhOUM7tWYIFpQ7Tmxfm9Ir8wYw2I=";
+  };
+
+  nativeBuildInputs = [
+    rustPlatform.cargoSetupHook
+    setuptools-rust
+  ] ++ (with rustPlatform; [ rust.cargo rust.rustc ]);
+
+  buildInputs = lib.optionals stdenv.isDarwin [ libiconv ];
+
+  propagatedBuildInputs = [
+    requests
+    regex
+    blobfile
+  ];
+
+  meta = with lib; {
+    description = "tiktoken is a fast BPE tokeniser for use with OpenAI's models.";
+    homepage = "https://github.com/openai/tiktoken";
+    license = licenses.asl20;
```

Comment:
I found a different license in the repository:
https://github.com/openai/tiktoken/blob/main/LICENSE